### DISCUSSION TITLE
@koshijs/core 中的 internal/validate 有用吗？

### DISCUSSION THREAD

**Discussion by DGCK81LNN** [koishijs/koishi#582] [Q&A] [answered]
我在写 locales 的时候发现 `core/internal/validate` 中似乎提供了检查指令参数的方法，但是我在调用自己的指令时，如果必填参数没有填写，`action` 的回调中还是会收到 `undefined`。好奇之下我看了一下我安装的 `@koishijs/core` 的 `.d.ts`，发现 `namespace internal` 是空的：
```typescript
export type ValidationField = 'authority' | 'usage' | 'timers';
declare namespace internal {
}
```
也就是说 `internal/validate` 除了这个 `type ValidationField` 以外什么也没导出出来！

**Comment by DGCK81LNN** [comment, marked as answer]
我明白了，原来[默认是不检查](https://github.com/koishijs/koishi/blob/7f904ee2f0bd13ec72a80fa3ebf3d7502f170de7/docs/api/core/context.md#ctxcommanddef-desc-config)：

> **`ctx.command(def, desc?, config?)`**
> - **config:** `CommandConfig` 指令的配置
>   - **checkUnknown:** `boolean` 是否对未知选项进行检测，默认为 `false`
>   - **checkArgCount:** `boolean` 是否对参数个数进行检测，默认为 `false`
>   - **authority:** `number` 最低调用权限，默认为 `1`
>   - **showWarning:** `boolean` 当小于最短间隔时是否进行提醒，默认为 `false`

这个指南里目前好像没有写，我之前一直没有注意到。

因为是内置插件，会自动注册，所以不需要导出。

https://github.com/koishijs/koishi/blob/037ad24e5e5a799140c3c643906d6db4bad636f5/packages/core/src/app.ts#L11

https://github.com/koishijs/koishi/blob/037ad24e5e5a799140c3c643906d6db4bad636f5/packages/core/src/app.ts#L83

**Reply by shigma** [reply]
是这样的x
